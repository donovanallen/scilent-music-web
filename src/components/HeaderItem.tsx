'use client';

import React from 'react';
import { IconType } from 'react-icons';

import { cn } from '@/lib/utils';

interface HeaderItemProps {
  title: string;
  image?: string;
  name: string;
  icon?: IconType;
  onClick?: () => void;
  className?: string;
  disabled?: boolean;
}

const HeaderItem: React.FC<HeaderItemProps> = ({
  title,
  image,
  name,
  icon: Icon,
  onClick,
  className,
  disabled = false,
}) => {
  // Prepare background image style if image prop is provided
  const backgroundImageStyle = image
    ? { backgroundImage: `url(${image})` }
    : {};

  return (
    // TODO update to bg image
    <button
      onClick={onClick}
      className={cn(
        'relative group  min-h-36 rounded-lg w-full overflow-hidden transition bg-black border-white border',
        'bg-cover bg-center', // Background size and position
        'semi-opaque-bg', // Custom class for semi-opaque background
        disabled ? 'opacity-30 cursor-not-allowed' : '',
        className,
      )}
      style={backgroundImageStyle}
    >
      <div className='flex flex-col justify-end w-full gap-y-2 px-4'>
        <h2 className='truncate'>{name}</h2>
        <div className='flex flex-col gap-y-1 w-full items-center self-center text-brand-dark'>
          <h4 className='subtitle'>{title}</h4>
          {Icon && <Icon size={20} />}
        </div>
      </div>
    </button>
  );
};

export default HeaderItem;
