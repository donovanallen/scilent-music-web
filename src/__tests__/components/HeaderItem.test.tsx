import { render } from '@testing-library/react';
import React from 'react';

import HeaderItem from '@/components/HeaderItem';

describe('HeaderItem', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<HeaderItem />);
    expect(asFragment()).toMatchSnapshot();
  });
});
