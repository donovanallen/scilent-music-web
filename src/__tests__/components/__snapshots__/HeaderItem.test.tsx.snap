// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderItem renders correctly 1`] = `
<DocumentFragment>
  <button
    class="relative group min-h-36 rounded-lg w-full overflow-hidden transition bg-black border-white border bg-cover bg-center semi-opaque-bg"
  >
    <div
      class="flex flex-col justify-end gap-y-2 px-4 w-full h-full"
    >
      <h2
        class="text-lg lg:text-2xl xl:text-3xl line-clamp-1"
      />
      <div
        class="flex flex-col gap-y-1 w-full items-center self-center text-brand-dark"
      >
        <h4
          class="subtitle text-xs lg:text-sm line-clamp-1"
        />
      </div>
    </div>
  </button>
</DocumentFragment>
`;
